src/main/java/org/teamresistance/frc/autonomous/AutoPingController.java

```diff
@@ -0,0 +1,32 @@
+package org.teamresistance.frc.autonomous;
+
+import org.teamresistance.frc.subsystem.Controller;
+import edu.wpi.first.wpilibj.PIDSourceType;
+import org.strongback.control.SoftwarePIDController;
+import org.teamresistance.frc.Pose;
+import org.teamresistance.frc.subsystem.Controller;
+import org.teamresistance.frc.util.SynchronousPID;
+
+
+/**
+ * Created by owner on 1/25/2017.
+ */
+public class AutoPingController implements Controller {
+  private SynchronousPID pingPID;
+  private static final double TOLERANCE = 2;
+  private static final double KP = 0.0;
+  private static final double KD = 0.0;
+  private static final double KI = 0.0;
+  private static final double FF = 0.0;
+  double targetDistance;
+
+  public AutoPingController(SynchronousPID pingPID) {
```

Comment:
`pingPID` should be instantiated within the constructor of this class, not passed from the outside. You want your parameter to just be the `targetDistance`, which would be assigned to the field. Now this controller is made to work with a specific target distance for the duration of its life.

```java
public AutoPingController(double targetDistanceInches) {
  this.targetDistanceInches = targetDistanceInches;
  this.pingPID = new SynchronousPID(...);
}
```

By making this class now **immutable** we avoid a whole class (pun not intended) of runtime bugs related to weird state (for example, Object A updates object B when object C is also referencing B and doesn't expect it to change, so C malfunctions or crashes). Object instances are cheap, so it's okay that we have to make a new `AutoPingController` every time we have a new target distance if it means we can avoid such bugs. 👌